src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -649,6 +649,10 @@ void FancyZones::OnDisplayChange(DisplayChangeType changeType) noexcept
         GUID currentVirtualDesktopId{};
         if (VirtualDesktopUtils::GetCurrentVirtualDesktopId(&currentVirtualDesktopId))
         {
+            if (m_currentDesktopId != GUID_NULL)
+            {
+                Trace::VirtualDesktopChanged();

```

Comment:
Personally, I would make check if previous desktop id differs from current desktop id (after asignment). Since event is called `VirtualDesktopChanged` and desktop is uniquely identified by desktop id, let's check it it changed.
Also this will aslo keep functionality that we want (don't report on boot-up since both will be `GUID_NULL`).

Reply:
Done